src/client.cc

```
@@ -1429,4 +1429,58 @@ RemoveObjectsResult Client::RemoveObjects(RemoveObjectsArgs args) {
+                      return Client::GetObject(std::move(args));
+                    });
+}
+
+std::future<ListObjectsResult> Client::ListObjectsAsync(ListObjectsArgs args) {
+  return std::async(std::launch::async,
+                    [this, args = std::move(args)]() mutable {
+                      return ListObjects(std::move(args));
+                    });
+}
+
+std::future<PutObjectResponse> Client::PutObjectAsync(PutObjectArgs args) {
```

Cppcheck (2.21.0)

[style] The function 'PutObjectAsync' is never used.

(unusedFunction)